src/app/components/inputs/select/index.tsx

```diff
@@ -1,81 +1,75 @@
-import React from 'react';
+import React, { useState } from 'react';
 import { FieldValues, useController } from 'react-hook-form';
 import { useTranslation } from 'react-i18next';
-import {
-  ChevronDownIcon,
-  FormControl,
-  FormControlError,
-  FormControlErrorText,
-  FormControlLabel,
-  FormControlLabelText,
-  Icon,
-  Select,
-  SelectBackdrop,
-  SelectContent,
-  SelectDragIndicator,
-  SelectDragIndicatorWrapper,
-  SelectIcon,
-  SelectInput,
-  SelectItem,
-  SelectPortal,
-  SelectTrigger,
-} from '@gluestack-ui/themed';
+import { FlatList, Modal } from 'react-native';
 
+import TextInput from 'src/app/components/inputs/text-input';
+import Text from 'src/app/components/text';
+
+import { BottomSheetContent, Option, Overlay } from './styles';
 import { SelectInputProps } from './types';
 
-// TODO: Redo without gluestack
 const ControlledSelectInput = <Form extends FieldValues>({
   controller,
-  inputProps = {},
-  formControlProps = {},
   options,
+  ...restOfProps
 }: SelectInputProps<Form>) => {
-  const { field, fieldState } = useController(controller);
+  const {
+    field: { onChange, onBlur },
+    fieldState: { error },
+  } = useController(controller);
   const { t } = useTranslation();
 
   const label = t(`inputs:label.${controller.name}`);
   const placeholder = t(`inputs:placeholder.${controller.name}`);
 
-  const handleSelect = (selectedValue: string) => {
-    field.onChange(selectedValue);
+  const [modalVisible, setModalVisible] = useState(false);
+  const [selectedLabel, setSelectedLabel] = useState('');
```

Comment:
Why do you have a state for control the select, why don't you use the react hook form to do this?

Reply:
the problem was related a to the form-config on the complete data section : it only accepts MALE and FEMALE as values to be send to the backend : 
```
const genderRegex = new RegExp(`^(MALE|FEMALE)$`, 'i');

gender: yup
    .string()
    .required(
      t('inputs:error.required', {
        field: t('inputs:label.gender').toLowerCase(),
      }),
    )
    .matches(
      genderRegex,
      t('inputs:error.genderValue', {
        male: t('common:gender.male').toLowerCase(),
        female: t('common:gender.female').toLowerCase(),
      }),
    ),
    ```
    so if we were seting the hook form value it would be populated as MALE/FEMALE, thats why i had to create a separate state for placing the label instead
    
    ```
    const handleSelect = (selectedValue: string, valueLabel: string) => {
    onChange(selectedValue);
    setSelectedLabel(valueLabel);
    setModalVisible(false);
  };
  ```
  
  Let me know if you think of a workaround, i reckon that this wont bother on the general usage, but i know its a bit fuzzy though